android-auto-app/src/main/java/com/mapbox/maps/testapp/auto/car/MapSession.kt

```diff
@@ -0,0 +1,89 @@
+package com.mapbox.maps.testapp.auto.car
+
+import android.Manifest.permission.ACCESS_FINE_LOCATION
+import android.content.Intent
+import android.content.pm.PackageManager.PERMISSION_GRANTED
+import android.content.res.Configuration
+import androidx.car.app.Screen
+import androidx.car.app.ScreenManager
+import androidx.car.app.Session
+import com.mapbox.maps.EdgeInsets
+import com.mapbox.maps.MapSurface
+import com.mapbox.maps.Style
+import com.mapbox.maps.extension.androidauto.initMapSurface
+import com.mapbox.maps.extension.style.layers.generated.skyLayer
+import com.mapbox.maps.extension.style.layers.properties.generated.SkyType
+import com.mapbox.maps.extension.style.sources.generated.rasterDemSource
+import com.mapbox.maps.extension.style.style
+import com.mapbox.maps.extension.style.terrain.generated.terrain
+import com.mapbox.maps.plugin.locationcomponent.location
+import com.mapbox.maps.testapp.auto.R
+
+/**
+ * Session class for the Mapbox Map sample app for Android Auto.
+ */
+class MapSession : Session() {
+  private lateinit var mapSurface: MapSurface
+  private val carCameraController = CarCameraController()
+
+  override fun onCreateScreen(intent: Intent): Screen {
+    val mapScreen = MapScreen(carContext)
+    initMapSurface(scrollListener = carCameraController) { surface ->
+      mapSurface = surface
+      carCameraController.init(
+        mapSurface,
+        EdgeInsets(
+          carContext.resources.getDimensionPixelSize(R.dimen.map_padding).toDouble(),
+          0.0,
+          0.0,
+          0.0
+        )
+      )
+      mapScreen.setMapCameraController(carCameraController)
+      loadStyle(surface)
+      initLocationComponent(surface)
+    }
+    return if (carContext.checkSelfPermission(ACCESS_FINE_LOCATION) != PERMISSION_GRANTED) {
+      carContext.getCarService(ScreenManager::class.java).push(mapScreen)
+      RequestPermissionScreen(carContext)
+    } else mapScreen
+  }
+
+  private fun loadStyle(surface: MapSurface) {
+    surface.getMapboxMap().loadStyle(
+      style(if (carContext.isDarkMode) Style.TRAFFIC_NIGHT else Style.TRAFFIC_DAY) {
+        +rasterDemSource(SOURCE) {
+          url(TERRAIN_URL_TILE_RESOURCE)
+          // 514 specifies padded DEM tile and provides better performance than 512 tiles.
+          tileSize(514)
```

Comment:
could be not specified explicitly already as it is fixed upstream

Reply:
Good to know that this is fixed, well I don't see the tileSize setting and the comment to be wrong 😄

Reply:
It's not wrong but you actually set same value as default.
Up to you of course